Add filter query parameters for servicegraph

SeriyBg opened the pull request:
Fix #6588 
Add the filter query parameters to filter by metric labels:
- destination_namespace
- destination_workload
- source_namespace
- source_workload

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6812#" title="Author self-approved">SeriyBg</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **geeknoid**

Assign the PR to them by writing `/assign @geeknoid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["geeknoid"]} -->

istio-testing commented:
Hi @SeriyBg. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

SeriyBg commented:
cc @douglas-reid 
Please review the PR when you have time. Also, consider that addition metric labels can be added for filtering servicegraph. If any other metrics labels are needed I can work on it as well. Thanks!

SeriyBg commented:
/assign @douglas-reid

zachgersh commented:
/ok-to-test

douglas-reid commented on addons/servicegraph/README.md:
```diff
@@ -37,6 +37,19 @@ All endpoints take these query parameters:
+- `destination_namespace` will filter the nodes and edges show to only 
+  those where the destination workload is in the given namespace.
+  
+- `destination_workload` will filter the nodes and edges show to only 
```
is this for showing only the direct calls for a given workload?  So, we would use this in combination with `source_workload` for showing the direct neighbors of a given workload ?

SeriyBg replied:
Yes, this parameter will show the direct call for a given workload. But the combination of these parameters as AND relation, so when we have `source_worload` and `destination_workload` will see only the direct call between these two workloads

douglas-reid commented on addons/servicegraph/promgen/promgen.go:
```diff
@@ -27,16 +27,22 @@ import (
 	"github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/prometheus/common/model"
 
+	"strings"
```
style nit: this should be in the top block of imports, next to `strconv`.

SeriyBg replied:
Done

douglas-reid commented:
@SeriyBg do you have any screenshots of the output from using these various options that you could share here?

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6812#" title="Author self-approved">SeriyBg</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **douglas-reid**

Assign the PR to them by writing `/assign @douglas-reid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["douglas-reid"]} -->

SeriyBg commented:
@douglas-reid I have addressed your comments, please have a look when you find the time. Let me know if there are other remarks or the rebase is needed

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6812#" title="Author self-approved">SeriyBg</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **douglas-reid**

If they are not already assigned, you can assign the PR to them by writing `/assign @douglas-reid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["douglas-reid"]} -->

SeriyBg commented:
/retest

douglas-reid commented on addons/servicegraph/promgen/promgen.go:
```diff
@@ -30,13 +31,17 @@ import (
 
-const reqsFmt = "sum(rate(istio_requests_total{reporter=\"server\"}[%s])) by (source_workload, destination_workload, source_app, destination_app)"
-const tcpFmt = "sum(rate(istio_tcp_received_bytes_total{reporter=\"server\"}[%s])) by (source_workload, destination_workload, source_app, destination_app)"
+const reqsFmt = "sum(rate(istio_requests_total{reporter=\"server\"%s}[%s])) by (source_workload, destination_workload, source_app, destination_app)"
```
`reporter=\"destination\"`

SeriyBg replied:
@douglas-reid are there any metric changes that happened so that `reporter=\"destination\"` must used instead of `reporter=\"server\"`?

douglas-reid replied:
https://github.com/istio/istio/pull/7034

the feedback i received was that `client` and `server` were a bit confusing (especially with inbound/outbound telemetry), so we updated to `source` and `destination` to (hopefully) be clearer.

SeriyBg replied:
@douglas-reid thanx. Have applied the changes and rebased to current master

douglas-reid commented:
/lgtm

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6812#issuecomment-409648507" title="LGTM">douglas-reid</a>*, *<a href="https://github.com/istio/istio/pull/6812#" title="Author self-approved">SeriyBg</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)~~ [douglas-reid]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
